internal/fields/validate.go

```diff
@@ -1266,23 +1260,19 @@ func ensureExpectedEventType(key string, values []string, definition FieldDefini
 	return nil
 }
 
-func valueToStringsSlice(value any) ([]string, error) {
+func valueToStringsSlice(value any) []string {
 	switch v := value.(type) {
 	case nil:
-		return nil, nil
+		return nil
 	case string:
-		return []string{v}, nil
+		return []string{v}
 	case []any:
 		var values []string
 		for _, e := range v {
-			s, ok := e.(string)
-			if !ok {
-				return nil, fmt.Errorf("expected string or array of strings")
-			}
-			values = append(values, s)
+			values = append(values, fmt.Sprintf("%v", e))
 		}
-		return values, nil
+		return values
 	default:
-		return nil, fmt.Errorf("expected string or array of strings")
+		return []string{fmt.Sprintf("%v", v)}
```

Comment:
I guess that if the value is an object (something different from string or []string), it would cause some validation error afterwards. Would that be the case?

Reply:
It will fail earlier, yes, I have moved the type validations before these checks https://github.com/elastic/elastic-package/pull/2065/files#diff-36d325d58b4ea8957abc63576d2e04d30ec8e58435ba06d442cb7923af063ad3L1018